File: 0036-valid-sudoku/0036-valid-sudoku.js
/**
 * @param {character[][]} board
 * @return {boolean}
 */
var isValidSudoku = function(board) {
    let validSet = new Set()
    for(let i = 0; i < 9; i++) {
        for(let j = 0; j < 9; j++) {
            // check cols
            // check boxes (3 * 3)
            // check rows
            // console.log(validSet)
            if(board[i][j] !== '.') {
                if(validSet.has(board[i][j] + " in row " + i) || 
                  validSet.has(board[i][j] + " in col " + j) || 
                  validSet.has(board[i][j] + " in box " + Math.floor(i/3) + ":" + Math.floor(j/3)))
                return false
                validSet.add(board[i][j] + " in row " + i)    
                validSet.add(board[i][j] + " in col " + j)    
                validSet.add(board[i][j] + " in box " + Math.floor(i/3) + ":" + Math.floor(j/3))    
            }
        }
    }
    return true
};

File: 0036-valid-sudoku/0036-valid-sudoku.java
class Solution {
    public boolean isValidSudoku(char[][] board) {
        Set<String> sudokuState = new HashSet<>();
        
        for(int i=0; i<board[0].length; i++) {
            for(int j=0; j<board.length; j++) {
                if(board[i][j] != '.') {
                    // check if set has the element
                    if(sudokuState.contains(board[i][j] + " in row " + i) || sudokuState.contains(board[i][j] + " in col " + j) || sudokuState.contains(board[i][j] + " in box " + i/3 + " : " + j/3))
                        return false;
                    
                    else {
                        sudokuState.add(board[i][j] + " in row " + i);
                        sudokuState.add(board[i][j] + " in col " + j);
                        sudokuState.add(board[i][j] + " in box " + i/3 + " : " + j/3);
                    }
                }
            }
        }
        
        return true;
    }
}